src/rgw/rgw_tracer.cc

```diff
@@ -0,0 +1,133 @@
+
+#include "rgw_tracer.h"
+
+using jConfig = jaegertracing::Config;
+using jLogger = jaegertracing::logging::Logger;
+
+namespace RGW_Tracing{
+
+    /*
+    //  Tracer
+    //
+    */
+    const std::shared_ptr<opentracing::Tracer> Tracer::noop_tracer = opentracing::MakeNoopTracer();
+    TracerConfiguration Tracer::tracer_conf;
+    
+    Tracer::Tracer(jConfig& conf):open_tracer(jaegertracing::Tracer::make(conf)){}
+    Tracer::Tracer(
+    std::string& service_name, 
+    const jConfig& conf, 
+    const std::shared_ptr<jLogger>& logger)
+    {
+        this->open_tracer = jaegertracing::Tracer::make(service_name, conf, logger);
+    }
+    Tracer::Tracer(std::string configFilePath){
+        auto configYAML = YAML::LoadFile(configFilePath);
+        auto config     = jaegertracing::Config::parse(configYAML);
+        this->open_tracer    = jaegertracing::Tracer::make(config);
+    }
+    Tracer::~Tracer(){
+        this->open_tracer->Close();
+    }
+    std::unique_ptr<Span> Tracer::start_trace(std::string trace_name){
+        if(is_enabled()) 
+            return std::unique_ptr<Span>(new Span(this->open_tracer->StartSpan(trace_name)));
+        else
+            return std::unique_ptr<Span>(new Span(this->noop_tracer->StartSpan(trace_name)));
+        
+    }
+    std::unique_ptr<Span> Tracer::start_span(std::string span_name, std::unique_ptr<Span>& parentSpan){
+        if(is_enabled()) 
+            return std::unique_ptr<Span>(new Span(this->open_tracer->StartSpan(span_name,{ parentSpan->get_reference() })));
+        else
+            return std::unique_ptr<Span>(new Span(this->noop_tracer->StartSpan(span_name)));
+    }
+    bool Tracer::is_enabled() const { return this->tracer_conf.is_enabled();}
+
+    /*
+    //  Span
+    //
+    */
+    Span::Span(std::unique_ptr<opentracing::Span> _span):span(_span.release()){};
+
+    Span::~Span(){ span->Finish();}
+
+    std::string Span::get_baggage(std::string key) const{
```

Comment:
what is Baggage here?  is this specific data?

Reply:
> what is Baggage here? is this specific data?
baggage is data that passes to child spans.

Reply:
what kind of information would you usually pass there?
is this only for the cases that `get_baggage()` is called in the child span?

Reply:
> what kind of information would you usually pass there? 
**I used it for debugging. I pass the trace's name through the child spans.** 
> is this only for the cases that `get_baggage()` is called in the child span?
**yes**